pages/register.tsx

```diff
@@ -145,515 +72,489 @@ export default function Register() {
   }
 
+  //disables submitting form on enter key press
+  function onKeyDown(keyEvent) {
+    if ((keyEvent.charCode || keyEvent.keyCode) === 13) {
+      keyEvent.preventDefault();
+    }
+  }
+
   return (
     <div className="flex flex-col flex-grow bg-white">
       <Head>
         <title>Hacker Registration</title>
         <meta name="description" content="Register for [HACKATHON NAME]" />
         <link rel="icon" href="/favicon.ico" />
       </Head>
-      <section id="jumbotron" className="p-2">
-        <div className="max-w-4xl py-6 mx-auto">
-          <div className="text-4xl  text-left">Hacker Registration</div>
-          <div className="text-1xl my-4 font-bold font-small text-left">
+
+      <section id="jumbotron" className="p-2 px-6">
+        <div className="max-w-4xl py-6 mx-auto flex flex-col items-center">
+          <div className="registrationTitle text-4xl font-bold text-center">
+            Hacker Registration
+          </div>
+          <div className="text-1xl my-4 font-bold font-small text-center">
             Please fill out the following fields. The application should take approximately 5
             minutes.
           </div>
         </div>
       </section>
-      <section id="registration" className="m-4">
-        <div className="max-w-4xl py-4 pt-8 mx-auto text-2xl font-bold text-left">General</div>
-        <div className="max-w-4xl py-4 mx-auto">
-          <form className="max-w-4xl mx-auto">
-            <label className="text-1xl my-4 font-bold font-small text-left">
-              *Enter your first name:
-              <br />
-              <input
-                className="border min-w-full pt-3 pb-3 text-grey-darkest px-5 bg-indigo-100 rounded-md"
-                placeholder="John"
-                type="text"
-                name="firstName"
-                autoComplete="given-name"
-                required
-                value={registrationData.user.firstName}
-                onChange={(e) => updateUserData(e)}
+
+      <section className="flex justify-center">
+        <Formik
+          initialValues={{
+            id: user?.id || '',
+            user: {
+              id: user?.id || '',
+              preferredEmail: user?.preferredEmail || '',
+              firstName: user?.firstName || '',
+              lastName: user?.lastName || '',
+              permissions: user?.permissions || ['hacker'],
+            },
+            age: 18,
+            gender: '',
+            race: '',
+            ethnicity: '',
+            university: '',
+            major: '',
+            studyLevel: '',
+            hackathonExperience: 0,
+            softwareExperience: 'Beginner',
+            heardFrom: 'Instagram',
+            size: 'S',
+            dietary: [],
+            accomodations: '',
+            github: '',
+            linkedin: '',
+            website: '',
+            companies: [],
+          }}
+          //validation
+          validate={(values) => {
```

Comment:
Need to simplify/abstractify validation checks. What will the process be if an organizer wants to add a custom question, or configure existing questions?

Reply:
Not sure on how to further simplify validation since you would just put the condition in which the values.[value] is invalid and set corresponding error message in errors.[value].